tests/fixtures/projects/test-package-type-fixer/pyproject.toml

```diff
@@ -0,0 +1,23 @@
+[build-system]
+requires = ["pdm-backend"]
+build-backend = "pdm.backend"
+
+[project]
+
+version = "0.0.1"
+dependencies = []
+name = "test-package-type-fixer"
+requires-python = ">=3.8"
+
+[tool.pdm.version]
+source = "file"
+path = "src/test_package_type_fixer/__init__.py"
+
+[tool.pdm]
+package-type = "application"
+
+[tool.pdm.dev-dependencies]
+
+dev = [
+    "black>=23.12.1"
```

Comment:
You'd better change to another package that is included in the mocked index, such as requests, to avoid unnecessary internet access.

Reply:
Forgot to mention, thank you for adding test cases.

Reply:
Oh, got it, let me change it.

Reply:
Will this work? 

```diff
dev = [
-    "black>=23.12.1"
+    "requests==2.19.1"
]
```

Or do I need to `project.add_dependencies(...`?

Reply:
yeah, that will work

Reply:
done :)